sdk/core/core/inc/az_span.h

```diff
@@ -135,161 +158,306 @@ AZ_NODISCARD az_span az_span_from_str(char* str);
+ * @param[in] source The int64 whose number is appended to the destination span as ASCII
+ * digits
+ * @param[out] out_span A pointer to an az_span that receives the span referring to the
+ * destination span with its length updated
+ * @return An #az_result value indicating the result of the operation:
+ *          #AZ_OK if successful
+ *          #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY if the destination is not big enough to
+ * contain the appended bytes
+ */
+AZ_NODISCARD az_result
+az_span_append_i64toa(az_span destination, int64_t source, az_span* out_span);
 
 /**
- * @brief a pair of az_span of bytes as a key and value
+ * @brief az_span_append_u64toa appends a uint64 as digit characters to the destination
+ * starting at the destination span's length.
  *
+ * @param[in] destination The az_span where the bytes should be appended to
+ * @param[in] source The uint64 whose number is appended to the destination span as ASCII
+ * digits
+ * @param[out] out_span A pointer to an az_span that receives the span referring to the
+ * destination span with its length updated
+ * @return An #az_result value indicating the result of the operation:
+ *          #AZ_OK if successful
+ *          #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY if the destination is not big enough to
+ * contain the appended bytes
  */
-typedef struct
-{
-  az_span key;
-  az_span value;
-} az_pair;
+AZ_NODISCARD az_result
+az_span_append_u64toa(az_span destination, uint64_t source, az_span* out_span);
 
-AZ_NODISCARD AZ_INLINE az_pair az_pair_init(az_span key, az_span value)
-{
-  return (az_pair){ .key = key, .value = value };
-}
+/**
+ * @brief az_span_append_dtoa appends a double as digit characters to the destination
+ * starting at the destination span's length
+ *
+ * @param[in] destination The az_span where the bytes should be appended to
+ * @param[in] source The double whose number is appended to the destination span as ASCII
+ * digits
+ * @param[out] out_span A pointer to an az_span that receives the span referring to the
+ * destination span with its length updated
+ * @return An #az_result value indicating the result of the operation:
+ *          #AZ_OK if successful
+ *          #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY if the destination is not big enough to
+ * contain the appended bytes
+ */
+AZ_NODISCARD az_result az_span_append_dtoa(az_span destination, double source, az_span* out_span);
 
-#define AZ_PAIR_NULL \
-  (az_pair) { .key = AZ_SPAN_NULL, .value = AZ_SPAN_NULL }
+/******************************  SPAN COPYING */
 
-AZ_NODISCARD AZ_INLINE az_pair az_pair_from_str(char* key, char* value)
+/**
+ * @brief az_span_set sets all the bytes of the destination span (up to its capacity) to \p
+ * fill.
+ *
+ * @param[in] destination The span whose bytes will be set to \p fill
+ * @param[in] fill The byte to be replicated within the destination span
+ */
+AZ_INLINE void az_span_set(az_span destination, uint8_t fill)
```

Comment:
We went back and forth on this one a few times. I think IoT might have wanted it; did you look to see if they are using it for anything? If not, then we should kill this API. 
Your questions are reasonable and I'd prefer not to think them through and just kill the API instead. 
The key client sample uses this to "erase" a secret so that it doesn't stay in memory long and to reset the buffer to a known state. So, if IOT doesn't use this, then lets kill the API entirely. Keyvault can just call memset directly.

Good catch!